internal/ingress/controller/store/store.go

```diff
@@ -773,3 +837,58 @@ func (s k8sStore) Run(stopCh chan struct{}) {
 		go wait.Until(s.checkSSLChainIssues, 60*time.Second, stopCh)
 	}
 }
+
+// getCurrentPod returns the current Pod based on environment namespace and name
+func (s k8sStore) getCurrentPod() *corev1.Pod {
+	ns := os.Getenv("POD_NAMESPACE")
+	name := os.Getenv("POD_NAME")
+
+	for _, i := range s.listers.Pod.List() {
+		pod := i.(*corev1.Pod)
+
+		if pod.GetNamespace() == ns &&
+			pod.GetName() == name {
+			return pod
+		}
+	}
+
+	return nil
+}
+
+// ListControllerPods returns a list of ingress-nginx controller Pods
+func (s k8sStore) ListControllerPods() []*corev1.Pod {
+	var pods []*corev1.Pod
+
+	curPod := s.getCurrentPod()
+	if curPod == nil {
+		return pods
+	}
+
+	for _, i := range s.listers.Pod.List() {
+		pod := i.(*corev1.Pod)
+
+		if !hasSameOwner(curPod, pod) {
+			continue
+		}
+
+		if pod.Status.Phase != corev1.PodRunning {
+			continue
+		}
+
+		pods = append(pods, pod)
+	}
+
+	return pods
+}
+
+// hasSameOwner returns true if both Pods share at least 1 owner
+func hasSameOwner(pod1, pod2 *corev1.Pod) bool {
```

Comment:
@aledbf do you know any better way of validating two Pods are deployed by the same `Deployment`?